adlfs/spec.py

```diff
@@ -1541,11 +1542,19 @@ async def _url(
         """
         container_name, blob, version_id = self.split_path(path)
 
+        if self.connection_string:
+            args_dict = parse_connection_string(self.connection_string)
```

Comment:
OK, I see what's going on here, looks good to me. One assumes the connected client also has this information, but since there's an upstream function to parse it, I don't see why we shouldn't use it.

Reply:
yep, thanks @martindurant for approving ... I hope maintainers will be able to take a look soon :)